src/operator/tensor/matrix_op-inl.h

```diff
@@ -1217,6 +1218,83 @@ void Slice(const nnvm::NodeAttrs& attrs,
   });
 }
 
+// slice the indptr of a csr
+struct SliceCsrIndPtr {
+  template<typename IType>
+  MSHADOW_XINLINE static void Map(int i, IType* out, const IType* in, IType* base) {
```

Comment:
Looks like there is no need to pass in base as an IType*?
Is "const IType" better since no need to dereference base every time in Map?